mathics/builtin/box/image.py

```diff
@@ -13,18 +32,80 @@ class ImageBox(BoxExpression):
     an Image object.
     """
 
+    def boxes_to_b64text(self, elements=None, **options):
+        contents, size = self.boxes_to_png(elements, **options)
+        encoded = base64.b64encode(contents)
+        encoded = b"data:image/png;base64," + encoded
+        return encoded, size
+
+    def boxes_to_png(self, elements=None, **options) -> Tuple:
+        """
+        returns a tuple with the set of bytes with a png representation of the image
+        and the scaled size.
+        """
+        image = self.elements[0] if elements is None else elements[0]
+
+        pixels = pixels_as_ubyte(image.color_convert("RGB", True).pixels)
+        shape = pixels.shape
+
+        width = shape[1]
+        height = shape[0]
+        scaled_width = width
+        scaled_height = height
+
+        # If the image was created from PIL, use that rather than
+        # reconstruct it from pixels which we can get wrong.
+        # In particular getting color-mapping info right can be
+        # tricky.
+        if hasattr(image, "pillow"):
+            pillow = deepcopy(image.pillow)
+        else:
+            pixels_format = "RGBA" if len(shape) >= 3 and shape[2] == 4 else "RGB"
+            pillow = PIL.Image.fromarray(pixels, pixels_format)
+
+        # if the image is very small, scale it up using nearest neighbour.
+        min_size = 128
+        if width < min_size and height < min_size:
+            scale = min_size / max(width, height)
+            scaled_width = int(scale * width)
+            scaled_height = int(scale * height)
+            pillow = pillow.resize(
+                (scaled_height, scaled_width), resample=PIL.Image.NEAREST
+            )
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+
+            stream = BytesIO()
+            pillow.save(stream, format="png")
+            stream.seek(0)
+            contents = stream.read()
+            stream.close()
+
+        return (contents, (scaled_width, scaled_height))
+
     def boxes_to_text(self, elements=None, **options):
         return "-Image-"
 
     def boxes_to_mathml(self, elements=None, **options):
-        if elements is None:
-            elements = self._elements
+        encoded, size = self.boxes_to_b64text(elements, **options)
         # see https://tools.ietf.org/html/rfc2397
-        return '<mglyph src="%s" width="%dpx" height="%dpx" />' % (
-            elements[0].get_string_value(),
-            elements[1].get_int_value(),
-            elements[2].get_int_value(),
-        )
+        return '<mglyph src="%s" width="%dpx" height="%dpx" />' % (encoded, *size)
```

Comment:
There is a little bit of magic here going on knowing that size has two items that exactly fills the two slots. (Automated checkers are probably not going to be able to figure this out, especially when the return type was that Tuple and one of the components of that was _also_ a tuple with two items).

Better would be to use an f-string and be explicit about the use of size: 

```python
return f'<mglyph src="{encoded}" width="{size[0]}px" height="{size[1]}px" />' 
```